### DISCUSSION TITLE
Is there any way to load a pre-generated SPIR-V module and run it with DPC++?

### DISCUSSION THREAD

**Discussion by YangQun1** [intel/llvm#5960] [Q&A]
Hi,

I have a pre-generated SPIR-V module, which contains a set of kernels in it. Now I want to load the module, extract kernels from it and launch kernels by using DPC++. Is there any way to do that?

Thanks

**Comment by alexbatashev** [comment]
If those are OpenCL kernels, then you should use interop API. If you somehow obtained DPC++ SPIR-V module, then you can't run kernels from it, because SPIR-V modules are not self-contained. You need integration header and image descriptor, and even then there's no public API, as the whole idea of running independent kernels conflicts with the single-source model of SYCL.

**Reply by YangQun1** [reply]
Hi, @alexbatashev  Thanks for your reply :)

The SPIR-V module is obtained in following way:
```shell
COMPILER_DIR=path-to-dpcpp-compile-daily-release/bin/
$COMPILER_DIR/clang -cc1 -triple spir kernel.cl -O2 -finclude-default-header -emit-llvm-bc -o kernel.bc
$COMPILER_DIR/llvm-spirv kernel.bc -o kernel.spv
```
I could load the module by using level zero, and run the kernel successfully. Could such module be loaded by using L0 backend interop API in DPC++?
